flowcraft/templates/process_tsv.py

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env python3
+
+"""
+Purpose
+-------
+This module is intended to process the output in tsv
+ to generate a report in json format.
+Expected input
+--------------
+The following variables are expected whether using NextFlow or the
+:py:func:`main` executor.
+- ``sample_id`` : Sample Identification string.
+- ``tsv``: tsv output.
+Generated output
+----------------
+- ``.report.json``: Data structure for the report
+Code documentation
+------------------
+"""
+
+import json
+import csv
+import os
+
+from flowcraft_utils.flowcraft_base import get_logger, MainWrapper
+
+__version__ = "1.0.1"
+__build__ = "05.10.2018"
+__template__ = "maxbin2-nf"
+
+logger = get_logger(__file__)
+
+if __file__.endswith(".command.sh"):
+    SAMPLE_ID = '$sample_id'
+    FILE = '$tsv'
+    logger.debug("Running {} with parameters:".format(
+        os.path.basename(__file__)))
+    logger.debug("SAMPLE_ID: {}".format(SAMPLE_ID))
+    logger.debug("FILE: {}".format(FILE))
+
+@MainWrapper
+def main(sample_id, tsv_file):
+
+    report_json = {
+        "tsvData": [{
+            "sample": sample_id,
+            "data": {}
+        }]
+    }
+
+    #web-app excepts a list with all the values in the table.
+    report_json["tsvData"][0]["data"]["MaxBin2"] = list(csv.reader(open(tsv_file), delimiter='\t'))
```

Comment:
Why didn't you set the `"MaxBin2"` directly on the literal dict definition. 

``` 
     report_json = {
         "tsvData": [{
             "sample": sample_id,
             "data": {
                 "MaxBin2": list(csv.reader(open(tsv_file), delimiter='\t'))
             }
         }]
     }
```

Reply:
To ease template extension to other processes. This way it's clear where the template needs to be edited.

Reply:
I'm not sure how this is clearer or easier to read than the literal definition?

Reply:
I prefer to make the literal definition general and populate the dictionary as needed. This way if the this template needs to be extended for other modules, the line that needs to be substituted by the appropriate method is separated. This has no impact on the execution time, but if you really dislike this, I can alter it.